barretenberg/cpp/src/barretenberg/vm2/common/protocol_contract_data.hpp

```diff
@@ -0,0 +1,28 @@
+
+// GENERATED FILE - DO NOT EDIT, RUN yarn generate in yarn-project/protocol-contracts
+#pragma once
+
+#include "barretenberg/vm2/common/aztec_constants.hpp"
+#include "barretenberg/vm2/common/aztec_types.hpp"
+
+#include <map>
+
+namespace bb::avm2 {
+
+using CanonicalAddress = AztecAddress;
+using DerivedAddress = AztecAddress;
+
+const std::map<CanonicalAddress, DerivedAddress> derived_addresses = {
```

Comment:
use `unordered_map` unless you need the insertion order to be respected.

only do the declaration in the hpp and have the actual map in a cpp like we do with https://github.com/AztecProtocol/aztec-packages/blob/e53d3d91401c4b0200652dae89be902978de857d/barretenberg/cpp/src/barretenberg/vm2/common/instruction_spec.hpp#L76

I'm surprised this even works